src/main/io/rcdevice_cam.c

```diff
@@ -0,0 +1,280 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "cms/cms.h"
+
+#include "fc/rc_controls.h"
+#include "fc/runtime_config.h"
+
+#include "io/beeper.h"
+#include "io/rcdevice_cam.h"
+
+#include "rx/rx.h"
+
+#ifdef USE_RCDEVICE
+
+#define IS_HI(X) (rcData[X] > 1750)
+#define IS_LO(X) (rcData[X] < 1250)
+#define IS_MID(X) (rcData[X] > 1250 && rcData[X] < 1750)
```

Comment:
1750 and 1250 does not fit in any category - it may possibly cause problems, it is beter to adjust bounds so that each value is one of HI/MID/LO

Reply:
1750 and 1250 is a number from osd.c and cms.c, perhaps they have been verified by the user, and  when user 3 position switch on transmitter, 1750 is hit at high, and 1250 is hit at low.

Reply:
I think this used to be `min_command` and `max_command` (originally used for stick commands). But it was probably found to be too clumsy to use for OSD since it requires the user to move the sticks almost to the end of the range.

Having these as magic numbers is not desirable. We should use constants (`OSD_MIN_COMMAND`, `OSD_MAX_COMMAND`?) for them now, and then change them to be configurable in a later pull request.

Reply:
Oh, I get your idea, I've define some macors instead of magic numbers, and adjust the range of the min\max\mid
```
#define FIVE_KEY_CABLE_JOYSTICK_MIN 1080
#define FIVE_KEY_CABLE_JOYSTICK_MAX 1920
#define FIVE_KEY_CABLE_JOYSTICK_MID_START 1350
#define FIVE_KEY_CABLE_JOYSTICK_MID_END 1650

#define IS_HI(X) (rcData[X] > FIVE_KEY_CABLE_JOYSTICK_MAX)
#define IS_LO(X) (rcData[X] < FIVE_KEY_CABLE_JOYSTICK_MIN)
#define IS_MID(X) (rcData[X] > FIVE_KEY_CABLE_JOYSTICK_MID_START && rcData[X] < FIVE_KEY_CABLE_JOYSTICK_MID_END)
```

i've test it with 5 key osd cable adapter, it's work fine.

Reply:
This should really be pulled out to some generic level, this is already duplicated in `osd.c` and 'cms.c`.

Is there a reason why different limits are used too?

Reply:
@DanNixon 

Agree, the IS_XX() marcos in ```osd.c```, ```cms.c``` and ```rcdevice_cam.c``` should pulled out to some generic level.

Why we are use different limits in rcdevice_cam.c, the idea is from @mikeller, because we are using joysticks to control the OSD menu in camera, most actions are require the user to move the joysticks almost to the end of the range. Use 1750 and 1250 as condition are right when using 3 or 2 position switch, because the 3 position alwasy hit on range.

Reply:
I agree with @DanNixon. What I meant with my comment above is that these defines should be shared between all of the areas where sticks are used to navigate menus (OSD / CMS / camera control / RunCam control / ...). If there are already defines for these values, then they should be used. Otherwise this is a good opportunity to introduce defines, and switch the other areas from magic numbers to these defines.

Reply:
OK, I get it.

Could I skip this modify suggestion in this PR? as I know, @DieHertz will unify them to avoid the codes duplication(both camera control and 5 key cable simulation in rcdevice) after this PR merged

Reply:
Yes, fine by me. As long as it's done and not forgotten. :-)